routes/beetles.py

```diff
@@ -0,0 +1,578 @@
+            di[era][model] = dict()
+            for si, sn_li in enumerate(sc_li[1:]):
+                scenario = dim_encodings["scenario"][si]
+                di[era][model][scenario] = dict()
+                for sni, risk_level in enumerate(sn_li):
+                    snowpack = dim_encodings["snowpack"][sni]
+                    di[era][model][scenario][snowpack] = dict()
+                    di[era][model][scenario][snowpack]["beetle-risk"] = dim_encodings[
+                        "beetle_risk"
+                    ][int(risk_level)]
+                    if beetle_percents is not None:
+                        di[era][model][scenario][snowpack][
+                            "percent-low-risk"
+                        ] = beetle_percents[ei + 1][mi + 1][si + 1][sni][1]
+                        di[era][model][scenario][snowpack][
+                            "percent-medium-risk"
+                        ] = beetle_percents[ei + 1][mi + 1][si + 1][sni][2]
+                        di[era][model][scenario][snowpack][
+                            "percent-high-risk"
+                        ] = beetle_percents[ei + 1][mi + 1][si + 1][sni][3]
+
+    return di
+
+
+def summarize_within_poly_marr(ds, poly_mask_arr, bandname="Gray"):
+    """Summarize a single Data Variable of a xarray.DataSet within a polygon.
+    Return the results as a nested dict.
+
+    Args:
+        ds (xarray.DataSet): DataSet with "Gray" as variable of
+            interest
+        poly_mask_arr (numpy.ma.core.MaskedArra): a masked array masking the cells intersecting
+            the polygon of interest
+        dim_encodings (dict): nested dictionary of thematic key value pairs that characterize the
+            data and map integer data coordinates to models, scenarios, eras, snowpacks, etc.
+        bandname (str): name of variable in ds, defaults to "Gray" for Rasdaman coverages where
+            the name is not given at ingest
+
+    Returns:
+        Nested dict of results for all non-X/Y axis combinations,
+    """
+
+    # operates on underlying DataArray
+    da = ds[bandname]
+
+    # get axis (dimension) names and make list of all coordinate combinations
+    all_dims = da.dims
+    dimnames = [dimname for dimname in all_dims if dimname not in ("X", "Y")]
+    iter_coords = list(
+        itertools.product(*[list(ds[dimname].values) for dimname in dimnames])
+    )
+
+    data_arr = []
+    for coords in iter_coords:
+        sel_di = {dimname: int(coord) for dimname, coord in zip(dimnames, coords)}
+        data_arr.append(da.sel(sel_di).values)
+    data_arr = np.array(data_arr)
+
+    # need to transpose the 2D spatial slices if X is the "rows" dimension
+    if all_dims.index("X") < all_dims.index("Y"):
+        data_arr = data_arr.transpose(0, 2, 1)
+
+    data_arr_mask = np.broadcast_to(poly_mask_arr.mask, data_arr.shape)
+    data_arr[data_arr_mask] = np.nan
+
+    # Adds one to each value to generate correct shape and
```

Comment:
Check this trailing `and` in the comment...is there anything else? The suspense! 😁